core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3104,8 +3105,18 @@ private static class ExtractImplementor extends AbstractRexCallImplementor {
 
     @Override Expression implementSafe(final RexToLixTranslator translator,
         final RexCall call, final List<Expression> argValueList) {
-      final TimeUnitRange timeUnitRange =
-          (TimeUnitRange) translator.getLiteralValue(argValueList.get(0));
+      // May need to convert the first argument from a String to a TimeUnitRange
+      final Object timeUnitRangeObj = translator.getLiteralValue(argValueList.get(0));
+      final TimeUnitRange timeUnitRange;
+      if (timeUnitRangeObj instanceof String) {
```

Comment:
During validation, it will test the value and return an error if not valid.
https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java#L3484

I have added a test for this error. It results in an error message like this:
```
java.sql.SQLException: Error while executing SQL "select date_part('foo', timestamp '2022-06-03 12:15:48.678')": From line 1, column 18 to line 1, column 22: 'foo' is not a valid time frame
```